packages/cli/test/integration/evaluation/test-definitions.api.test.ts

```diff
@@ -0,0 +1,296 @@
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.get('/evaluation/test-definitions');
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.count).toBe(1);
+		expect(resp.body.data.testDefinitions).toHaveLength(1);
+		expect(resp.body.data.testDefinitions[0].name).toBe('test');
+		expect(resp.body.data.testDefinitions[0].workflowId).toBe(workflowUnderTest.id);
+		expect(resp.body.data.testDefinitions[0].evaluationWorkflowId).toBe(null);
+	});
+
+	test('should retrieve test definitions list with pagination', async () => {
+		// Add a bunch of test definitions
+		const testDefinitions = [];
+
+		for (let i = 0; i < 15; i++) {
+			const newTest = Container.get(TestDefinitionRepository).create({
+				name: `test-${i}`,
+				workflow: { id: workflowUnderTest.id },
+			});
+			testDefinitions.push(newTest);
+		}
+
+		await Container.get(TestDefinitionRepository).save(testDefinitions);
+
+		// Fetch the first page
+		let resp = await authOwnerAgent.get('/evaluation/test-definitions?take=10');
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.count).toBe(15);
+		expect(resp.body.data.testDefinitions).toHaveLength(10);
+
+		// Fetch the second page
+		resp = await authOwnerAgent.get('/evaluation/test-definitions?take=10&skip=10');
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.count).toBe(15);
+		expect(resp.body.data.testDefinitions).toHaveLength(5);
+	});
+});
+
+describe('GET /evaluation/test-definitions/:id', () => {
+	test('should retrieve test definition', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.get(`/evaluation/test-definitions/${newTest.id}`);
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.name).toBe('test');
+		expect(resp.body.data.workflowId).toBe(workflowUnderTest.id);
+		expect(resp.body.data.evaluationWorkflowId).toBe(null);
+	});
+
+	test('should return 404 for non-existent test definition', async () => {
+		const resp = await authOwnerAgent.get('/evaluation/test-definitions/123');
+
+		expect(resp.statusCode).toBe(404);
+	});
+
+	test('should retrieve test definition with evaluation workflow', async () => {
+		const newTest = Container.get(TestDefinitionRepository).create({
+			name: 'test',
+			workflow: { id: workflowUnderTest.id },
+			evaluationWorkflow: { id: evaluationWorkflow.id },
+		});
+		await Container.get(TestDefinitionRepository).save(newTest);
+
+		const resp = await authOwnerAgent.get(`/evaluation/test-definitions/${newTest.id}`);
+
+		expect(resp.statusCode).toBe(200);
+		expect(resp.body.data.name).toBe('test');
+		expect(resp.body.data.workflowId).toBe(workflowUnderTest.id);
+		expect(resp.body.data.evaluationWorkflowId).toBe(evaluationWorkflow.id);
+	});
```

Comment:
Maybe a test case so that user can't get test definition they don't have access to?